
*** Tallinn:MacPerl_Src:perl:ext:Bit-Vector-5.6-Dev:examples:primes.pl	Sun Mar  7 16:00:02 1999
--- Tallinn:MacPerl_Src:perl:ext:Bit-Vector-5.6-bin-1-MacOS:examples:primes.pl	Sun Mar  7 16:16:08 1999
***************
*** 19,22 ****
  
! $limit = $ARGV[0];
  
  $set = Bit::Vector->new($limit+1);
--- 19,38 ----
  
! $limit = 0;
  
+ if (-t STDIN)
+ {
+     while ($limit < 16)
+     {
+         print "\nPlease enter an upper limit (>15): ";
+         $limit = <STDIN>;
+         if ($limit =~ /^\s*(\d+)\s*$/) { $limit = $1; } else { $limit = 0; }
+     }
+     print "\n";
+ }
+ else
+ {
+     $limit = 100;
+     print "\nRunning in batch mode - using $limit as upper limit.\n\n";
+ }
+ 
  $set = Bit::Vector->new($limit+1);
***************
*** 61,63 ****
  {
! 	print "Elapsed time = ", ($stop-$start), " seconds\n";
  }
--- 77,102 ----
  {
!     my($flag) = 0;
!     my($sec,$min,$hour,$year,$yday) = (gmtime($stop - $start))[0,1,2,5,7];
!     $year -= 70;
!     print "Elapsed time: ";
!     if ($year > 0)
!     {
!         printf("%d year%s ", $year, ($year!=1)?"s":"");
!         $flag = 1;
!     }
!     if (($yday > 0) || $flag)
!     {
!         printf("%d day%s ", $yday, ($yday!=1)?"s":"");
!         $flag = 1;
!     }
!     if (($hour > 0) || $flag)
!     {
!         printf("%d hour%s ", $hour, ($hour!=1)?"s":"");
!         $flag = 1;
!     }
!     if (($min > 0) || $flag)
!     {
!         printf("%d minute%s ", $min, ($min!=1)?"s":"");
!     }
!     printf("%d second%s.\n\n", $sec, ($sec!=1)?"s":"");
  }

