
*** Tallinn:MacPerl_Src:perl:ext:Bit-Vector-5.6-Dev:Vector.xs	Sun Mar  7 12:40:48 1999
--- Tallinn:MacPerl_Src:perl:ext:Bit-Vector-5.6-bin-1-MacOS:Vector.xs	Sun Mar  7 16:16:09 1999
***************
*** 231,233 ****
      {
!         if ((address = BitVector_Create(size,trueval)) != NULL)
          {
--- 231,233 ----
      {
!         if ((address = BitVector_Create(size,true)) != NULL)
          {
***************
*** 264,266 ****
          {
!             if ((address = BitVector_Create(size,falseval)) != NULL)
              {
--- 264,266 ----
          {
!             if ((address = BitVector_Create(size,false)) != NULL)
              {
***************
*** 307,309 ****
          {
!             if ((address = BitVector_Create(size,falseval)) != NULL)
              {
--- 307,309 ----
          {
!             if ((address = BitVector_Create(size,false)) != NULL)
              {
***************
*** 350,352 ****
          {
!             if ((address = BitVector_Create(size,falseval)) != NULL)
              {
--- 350,352 ----
          {
!             if ((address = BitVector_Create(size,false)) != NULL)
              {
***************
*** 393,395 ****
          {
!             if ((address = BitVector_Create(size,falseval)) != NULL)
              {
--- 393,395 ----
          {
!             if ((address = BitVector_Create(size,false)) != NULL)
              {
***************
*** 526,528 ****
      }
!     if ((address = BitVector_Create(bits,falseval)) != NULL)
      {
--- 526,528 ----
      }
!     if ((address = BitVector_Create(bits,false)) != NULL)
      {
***************
*** 995,997 ****
  
! booltype
  BitVector_is_empty(reference)
--- 995,997 ----
  
! boolean
  BitVector_is_empty(reference)
***************
*** 1013,1015 ****
  
! booltype
  BitVector_is_full(reference)
--- 1013,1015 ----
  
! boolean
  BitVector_is_full(reference)
***************
*** 1031,1033 ****
  
! booltype
  BitVector_equal(Xref,Yref)
--- 1031,1033 ----
  
! boolean
  BitVector_equal(Xref,Yref)
***************
*** 1369,1371 ****
  
! booltype
  BitVector_bit_flip(reference,index)
--- 1369,1371 ----
  
! boolean
  BitVector_bit_flip(reference,index)
***************
*** 1399,1401 ****
  
! booltype
  BitVector_bit_test(reference,index)
--- 1399,1401 ----
  
! boolean
  BitVector_bit_test(reference,index)
***************
*** 1440,1442 ****
      N_int idx;
!     booltype b;
  
--- 1440,1442 ----
      N_int idx;
!     boolean b;
  
***************
*** 1445,1447 ****
          if ( BIT_VECTOR_SCALAR(index,N_int,idx) &&
!              BIT_VECTOR_SCALAR(bit,booltype,b) )
          {
--- 1445,1447 ----
          if ( BIT_VECTOR_SCALAR(index,N_int,idx) &&
!              BIT_VECTOR_SCALAR(bit,boolean,b) )
          {
***************
*** 1467,1469 ****
      BitVector_Address address;
!     booltype b;
  
--- 1467,1469 ----
      BitVector_Address address;
!     boolean b;
  
***************
*** 1471,1473 ****
      {
!         if ( BIT_VECTOR_SCALAR(bit,booltype,b) )
          {
--- 1471,1473 ----
      {
!         if ( BIT_VECTOR_SCALAR(bit,boolean,b) )
          {
***************
*** 1489,1491 ****
      BitVector_Address address;
!     booltype b;
  
--- 1489,1491 ----
      BitVector_Address address;
!     boolean b;
  
***************
*** 1493,1495 ****
      {
!         if ( BIT_VECTOR_SCALAR(bit,booltype,b) )
          {
--- 1493,1495 ----
      {
!         if ( BIT_VECTOR_SCALAR(bit,boolean,b) )
          {
***************
*** 1503,1505 ****
  
! booltype
  BitVector_lsb(reference)
--- 1503,1505 ----
  
! boolean
  BitVector_lsb(reference)
***************
*** 1521,1523 ****
  
! booltype
  BitVector_msb(reference)
--- 1521,1523 ----
  
! boolean
  BitVector_msb(reference)
***************
*** 1539,1541 ****
  
! booltype
  BitVector_rotate_left(reference)
--- 1539,1541 ----
  
! boolean
  BitVector_rotate_left(reference)
***************
*** 1557,1559 ****
  
! booltype
  BitVector_rotate_right(reference)
--- 1557,1559 ----
  
! boolean
  BitVector_rotate_right(reference)
***************
*** 1575,1577 ****
  
! booltype
  BitVector_shift_left(reference,carry)
--- 1575,1577 ----
  
! boolean
  BitVector_shift_left(reference,carry)
***************
*** 1583,1585 ****
      BitVector_Address address;
!     booltype c;
  
--- 1583,1585 ----
      BitVector_Address address;
!     boolean c;
  
***************
*** 1587,1589 ****
      {
!         if ( BIT_VECTOR_SCALAR(carry,booltype,c) )
          {
--- 1587,1589 ----
      {
!         if ( BIT_VECTOR_SCALAR(carry,boolean,c) )
          {
***************
*** 1599,1601 ****
  
! booltype
  BitVector_shift_right(reference,carry)
--- 1599,1601 ----
  
! boolean
  BitVector_shift_right(reference,carry)
***************
*** 1607,1609 ****
      BitVector_Address address;
!     booltype c;
  
--- 1607,1609 ----
      BitVector_Address address;
!     boolean c;
  
***************
*** 1611,1613 ****
      {
!         if ( BIT_VECTOR_SCALAR(carry,booltype,c) )
          {
--- 1611,1613 ----
      {
!         if ( BIT_VECTOR_SCALAR(carry,boolean,c) )
          {
***************
*** 1686,1688 ****
              {
!                 BitVector_Insert(address,off,cnt,trueval);
              }
--- 1686,1688 ----
              {
!                 BitVector_Insert(address,off,cnt,true);
              }
***************
*** 1715,1717 ****
              {
!                 BitVector_Delete(address,off,cnt,trueval);
              }
--- 1715,1717 ----
              {
!                 BitVector_Delete(address,off,cnt,true);
              }
***************
*** 1725,1727 ****
  
! booltype
  BitVector_increment(reference)
--- 1725,1727 ----
  
! boolean
  BitVector_increment(reference)
***************
*** 1743,1745 ****
  
! booltype
  BitVector_decrement(reference)
--- 1743,1745 ----
  
! boolean
  BitVector_decrement(reference)
***************
*** 1761,1763 ****
  
! booltype
  BitVector_add(Xref,Yref,Zref,carry)
--- 1761,1763 ----
  
! boolean
  BitVector_add(Xref,Yref,Zref,carry)
***************
*** 1775,1777 ****
      BitVector_Address Zadr;
!     booltype c;
  
--- 1775,1777 ----
      BitVector_Address Zadr;
!     boolean c;
  
***************
*** 1781,1783 ****
      {
!         if ( BIT_VECTOR_SCALAR(carry,booltype,c) )
          {
--- 1781,1783 ----
      {
!         if ( BIT_VECTOR_SCALAR(carry,boolean,c) )
          {
***************
*** 1797,1799 ****
  
! booltype
  BitVector_subtract(Xref,Yref,Zref,carry)
--- 1797,1799 ----
  
! boolean
  BitVector_subtract(Xref,Yref,Zref,carry)
***************
*** 1811,1813 ****
      BitVector_Address Zadr;
!     booltype c;
  
--- 1811,1813 ----
      BitVector_Address Zadr;
!     boolean c;
  
***************
*** 1817,1819 ****
      {
!         if ( BIT_VECTOR_SCALAR(carry,booltype,c) )
          {
--- 1817,1819 ----
      {
!         if ( BIT_VECTOR_SCALAR(carry,boolean,c) )
          {
***************
*** 2211,2213 ****
              {
!                 BitVector_Word_Insert(address,off,cnt,trueval);
              }
--- 2211,2213 ----
              {
!                 BitVector_Word_Insert(address,off,cnt,true);
              }
***************
*** 2240,2242 ****
              {
!                 BitVector_Word_Delete(address,off,cnt,trueval);
              }
--- 2240,2242 ----
              {
!                 BitVector_Word_Delete(address,off,cnt,true);
              }
***************
*** 2747,2749 ****
  
! booltype
  Set_subset(Xref,Yref)
--- 2747,2749 ----
  
! boolean
  Set_subset(Xref,Yref)

